guice/src/test/java/org/jdbi/v3/guice/TestJdbiBinder.java

```diff
@@ -0,0 +1,212 @@
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class TestJdbiBinder {
+
+    @Rule
+    public SingleInstancePostgresRule pg = EmbeddedPostgresRules.singleInstance();
+
+    @Inject
+    @BinderTest
+    public Jdbi jdbi = null;
+
+    @Before
+    public void setUp() {
+        Module testModule = new AbstractJdbiModule(BinderTest.class) {
+            @Override
+            protected void configureJdbi() {
+                bindPlugin().toInstance(PostgresPlugin.noUnqualifiedHstoreBindings());
+                bindPlugin().toInstance(new SqlObjectPlugin());
+
+                // pass testColumnMapper
+                bindColumnMapper().to(MyStringColumnMapper.class).in(Scopes.SINGLETON);
+
+                // pass testRowMapper
+                bindRowMapper().to(BinderTestRow.Mapper.class).in(Scopes.SINGLETON);
+            }
+        };
+
+        Injector inj = Guice.createInjector(Stage.PRODUCTION,
+            testModule,
+            binder -> binder.bind(DataSource.class).annotatedWith(BinderTest.class).toInstance(pg.getEmbeddedPostgres().getPostgresDatabase()),
+            Binder::disableCircularProxies,
+            Binder::requireExplicitBindings,
+            Binder::requireExactBindingAnnotations,
+            Binder::requireAtInjectOnConstructors);
+
+        inj.injectMembers(this);
+
+        assertNotNull(jdbi);
+        createDb();
+    }
+
+    private void createDb() {
+        jdbi.inTransaction(h -> {
+            h.execute("DROP TABLE IF EXISTS binder_test");
+            h.execute("CREATE TABLE binder_test (i INT, u UUID, s VARCHAR, t timestamp with time zone default current_timestamp)");
+            return null;
+        });
+
+        for (int c = 0; c < 100; c++) {
+            final int i = c;
+            final String s = "xxx" + c;
+            final UUID u = UUID.randomUUID();
+
+            int result = jdbi.withExtension(Dao.class, dao -> dao.createRow(i, u, s));
+            assertEquals(1, result);
+        }
+
+    }
+
+    @Test
+    public void testSimple() {
+        assertEquals(100, jdbi.withExtension(Dao.class, Dao::countRows).intValue());
+    }
+
+    @Test
+    public void testColumnMapper() {
+        List<MyString> uuids = jdbi.withExtension(Dao.class, Dao::getMyStrings);
+        assertEquals(100, uuids.size());
+    }
+
+    @Test
+    public void testRowMapper() {
+        List<BinderTestRow> rows = jdbi.withExtension(Dao.class, Dao::getRows);
+
+        for (int i = 0; i < 100; i++) {
+            assertEquals(i, rows.get(i).getI());
+        }
+    }
+
+    //
+
+    @Retention(RUNTIME)
+    @Target({FIELD, PARAMETER, METHOD})
+    @Qualifier
+    public @interface BinderTest {}
+
+    public interface Dao {
+
+        @SqlUpdate("INSERT INTO binder_test (i, u, s) VALUES (:i, :u, :s)")
+        int createRow(@Bind("i") int i, @Bind("u") UUID u, @Bind("s") String s);
```

Comment:
Are these `@Bind` declarations necessary? this module looks to have parameter names included

Reply:
probably not. However, I feel that in this case I probably replace the annotations with comments saying "This works because it is compiled with parameter names included".